File: bundled/tool/xray/utils/colons_test.py
import pytest

from .colons import escape_colons, unescape_colons


@pytest.mark.parametrize(
    "unescaped,escaped",
    [
        ("filename.py", "filename.py"),
        ("file:with:colons.py", r"file\:with\:colons.py"),
        ("filewithdouble::colons.py", r"filewithdouble\:\:colons.py"),
    ],
)
def test_colon_escape(
    unescaped: str,
    escaped: str,
):
    assert escape_colons(unescaped) == escaped
    assert unescape_colons(escaped) == unescaped

File: src/test/suite/colons.test.ts
import * as assert from 'assert';
import { escapeColons, unescapeColons } from '../../Colon';

suite('Colon Test Suite', () => {
    const testCases: { unescaped: string; escaped: string }[] = [
        { unescaped: 'filename.py', escaped: 'filename.py' },
        { unescaped: 'file:with:colons.py', escaped: 'file\\:with\\:colons.py' },
        { unescaped: 'filewithdouble::colons.py', escaped: 'filewithdouble\\:\\:colons.py' },
    ];

    testCases.forEach(({ unescaped, escaped }) => {
        test(`should escape and unescape "${unescaped}" correctly`, () => {
            assert.strictEqual(escapeColons(unescaped), escaped);
            assert.strictEqual(unescapeColons(escaped), unescaped);
        });
    });
});
